doc/develop/test/pytest.rst

```diff
@@ -137,6 +171,135 @@ For example, here is a test with a fixture ``mcumgr``
       mcumgr.reset_device()
       # continue test scenario, check version etc.
 
+Classes
+*******
+
+DeviceAdapter
+=============
+
+.. autoclass:: twister_harness.DeviceAdapter
+
+   .. automethod:: launch
+
+   .. automethod:: connect
+
+   .. automethod:: readline
+
+   .. automethod:: readlines
+
+   .. automethod:: readlines_until
+
+   .. automethod:: write
+
+   .. automethod:: disconnect
+
+   .. automethod:: close
+
+.. _shell_class:
+
+Shell
+=====
+
+.. autoclass:: twister_harness.Shell
+
+   .. automethod:: exec_command
+
+   .. automethod:: wait_for_prompt
+
+
+Exemplary pytest tests in Zephyr project
+****************************************
+
+1. Shell demo - :zephyr_file:`samples/subsys/testsuite/pytest/shell`
+#. MCUmgr tests - :zephyr_file:`tests/boot/with_mcumgr`
+#. LwM2M tests - :zephyr_file:`tests/net/lib/lwm2m/interop`
+#. GDB stub test - :zephyr_file:`tests/subsys/debug/gdbstub`
+
+
+FAQ
+***
+
+1. **How to flash/run application only once per pytest session?**
```

Comment:
I don't think numbering is necessary + it's prob better to actually keep the hierarchy using a proper heading
```suggestion
How to flash/run application only once per pytest session?
==========================================================
```